nautobot/project-static/js/forms.js

```diff
@@ -156,157 +156,162 @@ function initializeColorPicker(context, dropdownParent=null){
+                    var results = data.results;
+
+                    results = results.reduce((results,record,idx) => {
+                        record.text = record[element.getAttribute('display-field')] || record.name;
+                        if (record._depth) {
+                            // Annotate hierarchical depth for MPTT objects
+                            record.text = '--'.repeat(record._depth) + ' ' + record.text;
+                        }
+
+                        record.id = record[element.getAttribute('value-field')] || record.id;
+                        if(element.getAttribute('disabled-indicator') && record[element.getAttribute('disabled-indicator')]) {
+                            // The disabled-indicator equated to true, so we disable this option
+                            record.disabled = true;
+                        }
+
+                        if( record.group !== undefined && record.group !== null && record.site !== undefined && record.site !== null ) {
+                            results[record.site.name + ":" + record.group.name] = results[record.site.name + ":" + record.group.name] || { text: record.site.name + " / " + record.group.name, children: [] };
+                            results[record.site.name + ":" + record.group.name].children.push(record);
+                        }
+                        else if( record.group !== undefined && record.group !== null ) {
+                            results[record.group.name] = results[record.group.name] || { text: record.group.name, children: [] };
+                            results[record.group.name].children.push(record);
+                        }
+                        else if( record.site !== undefined && record.site !== null ) {
+                            results[record.site.name] = results[record.site.name] || { text: record.site.name, children: [] };
+                            results[record.site.name].children.push(record);
+                        }
+                        else if ( (record.group !== undefined || record.group == null) && (record.site !== undefined || record.site === null) ) {
+                            results['global'] = results['global'] || { text: 'Global', children: [] };
+                            results['global'].children.push(record);
+                        }
+                        else {
+                            results[idx] = record;
+                        }
+
+                        return results;
+                    },Object.create(null));
+
+                    results = Object.values(results);
+
+                    // Handle the null option, but only add it once
+                    if (element.getAttribute('data-null-option') && data.previous === null) {
+                        results.unshift({
+                            id: 'null',
+                            text: element.getAttribute('data-null-option')
+                        });
```

Comment:
Does this need logic to handle the case where there's no `data-null-option`?

Reply:
`element.getAttribute('data-null-option')` should return undefined. I haven't modified any of this code, it just got indented because of the change from
`this_context.find('.nautobot-select2-api').select2({ #...`
to
`this_context.find('.nautobot-select2-api').each(function(){ #...`

Reply:
I meant more do we need to do anything to `results` in the case where there is no `data-null-option` to handle the `--------` case?

Reply:
I think that would address a different UX in always allowing you to select the null option from the list, but that might not be useful for required fields.